src/directLine.ts

```diff
@@ -496,10 +496,16 @@ export class DirectLine implements IBotConnection {
                     }
                 })
                 .map(ajaxResponse => {
-                    konsole.log("getSessionId response: " + ajaxResponse.response.sessionId);
-                    return ajaxResponse.response.sessionId as string;
+                    if (ajaxResponse && ajaxResponse.response && ajaxResponse.response.sessionId) {
+                        konsole.log("getSessionId response: " + ajaxResponse.response.sessionId);
+                        return ajaxResponse.response.sessionId as string;
+                    }
+                    return '';
+                })
+                .catch(error => {
+                    konsole.log("getSessionId error: " + error.status);
+                    return Observable.of('');
```

Comment:
this is ok...getSessionId only works for certain DirectLine tokens and returns 403s otherwise, which this thing throws into errors. In those cases, DirectLine can return no session id as WebChat will then pass on the empty string as part of the auth sequence and the user will get the magic code verification...which is what we want. All of the getSessionId non-200 responses are logged in our service and I have a dashboard of these.

So I think returning empty string here is fine.